lib/tldr.js

```diff
@@ -13,188 +13,200 @@ const render = require('./render');
 const index = require('./index');
 const exit = process.exit;
 
-exports.list = (singleColumn) => {
-  let os = platform.getPreferredPlatformFolder();
-  index.commandsFor(os)
-    .then((commands) => {
-      printPages(commands, singleColumn);
-    });
-};
+class Tldr {
+  constructor(config) {
+    // TODO: replace this with a private field when it reaches enough maturity
+    // https://github.com/tc39/proposal-class-fields#private-fields
+    this.config = config;
+    this.cache = new Cache(this.config);
+  }
 
-exports.listAll = (singleColumn) => {
-  index.commands()
-    .then((commands) => {
-      printPages(commands, singleColumn);
-    });
-};
-
-exports.get = (commands, options) => {
-  printBestPage(commands.join('-'), options);
-};
-
-exports.random = (options) => {
-  let os = platform.getPreferredPlatformFolder();
-  index.commandsFor(os)
-    .then((pages) => {
-      if (pages.length === 0) {
-        console.error(messages.emptyCache());
+  list(singleColumn) {
+    let os = platform.getPreferredPlatformFolder(this.config);
+    index.commandsFor(os)
+      .then((commands) => {
+        this.printPages(commands, singleColumn);
+      });
+  }
+
+  listAll(singleColumn) {
+    index.commands()
+      .then((commands) => {
+        this.printPages(commands, singleColumn);
+      });
+  }
+
+  get(commands, options) {
+    this.printBestPage(commands.join('-'), options);
+  }
+
+  random(options) {
+    let os = platform.getPreferredPlatformFolder(this.config);
+    index.commandsFor(os)
+      .then((pages) => {
+        if (pages.length === 0) {
+          console.error(messages.emptyCache());
+          exit(1);
+        }
+        let page = sample(pages);
+        console.log('PAGE', page);
+        this.printBestPage(page, options);
```

Comment:
Because this is a reference to `this` inside a `.then()`, does the arrow function need to be bound like this?

```js
// ....
.then((() => {
    // ....
}).bind(this);
```

Sometimes it does and sometimes it doesn't - I'm actually rather unclear on how it works here.

In some situations I find that `this` refers to the global scope instead of the class instance, leading to errors.

Reply:
It doesn't. `this` inside an arrow function does not redefine it. It just uses the parent `this`. https://exploringjs.com/es6/ch_arrow-functions.html#sec_traditional-functions-bad-non-methods

Reply:
Oooh, I didn't know that arrow functions have different scoping rules. I'll have to play around with  that. Fascinating!